image.go

```diff
@@ -51,6 +52,10 @@ type Image interface {
 	IsUnpacked(context.Context, string) (bool, error)
 	// ContentStore provides a content store which contains image blob data
 	ContentStore() content.Store
+	// SetGPGClient sets the GPG client to use when decrypting the image
+	SetGPGClient(gpgClient encryption.GPGClient)
```

Comment:
These interface methods are _way_ to specific for the image. Notice the other methods aren't specific to an implementation or specific image format. These should probably be set through the client or as part of an option set.

Reply:
I think by now this could be a map[string]string where we pass the different types of decryption keys (gpg, pem/der keys) and other material (x509s).